### ISSUE TITLE
Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by brendanw** [npm/npm#19068]
#### I'm opening this issue because:

  - [ X] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [X ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
When running "npm install csv-parse --save" I am repeatedly getting the below error message:

npm ERR! Cannot read property '0' of undefined

#### How can the CLI team reproduce the problem?
I am using node 8.6.0 and npm 5.3.0. 

Debug log is available at:
https://gist.github.com/brendanw/73278a49b19e44ed86d6e0038f3417e0

### supporting information:

 - `npm -v` prints: 5.5.1
 - `node -v` prints: 8.9.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [X] I deploy to a PaaS (Triton, Heroku).

**Comment by gillstrom** [comment]
Could be the same issue as in https://github.com/npm/npm/issues/17858

**Comment by brendanw** [comment]
Yup, appears to be the same issue. Downgrading to npm 5.0.4 cleared the issue.
